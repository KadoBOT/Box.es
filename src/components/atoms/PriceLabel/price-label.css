.price-label {
  padding: 10px 5px;
  font-weight: 700;
  position: absolute;
  right: 0;
  text-align: center;
  top: 10px;
  background: rgba(0, 0, 0, 0.72);
  color: white;
  font-size: 41px;
}

.price-label__price__coin {
  font-size: 17px;
  vertical-align: text-top;
  line-height: 40px;
  margin-right: -10px;
}
