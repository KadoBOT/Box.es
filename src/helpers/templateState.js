export default {
  "isFetching": false,
  "name": "Diamond Select Marvel Select: Juggernaut Action Figure",
  "description": "A Diamond Select Release! Sculpted by Gentle Giant! The ranks of the best-selling Marvel Select line expand to include one of the most deadly mutants ever, the unstoppable Juggernaut. Sculpted by Gentle Giant and standing a massive 7\" tall, Juggernaut is the latest in the, in scale, over sized figures from the Marvel Select line. Charles Xavier's half brother is ready to join Magneto in the fight against the X-Men and anyone else that stands in their way. Featuring Marvel Selects standard 16 points of articulation, Juggernaut, and the sans-helmet chase variant figure, are the textbook definition of Unstoppable!",
  "price": 22.43,
  "location_details": "Genosha",
  "currency": {
    "symbol": "$",
  },
  "likes_count": 80,
  "user": {
    "nickname": "xavier",
    "name": "Charles Xavier",
    "avatar": {
      "small_url": "http://static.comicvine.com/uploads/original/7/71941/1489898-charles_xavier.jpg",
    },
  },
  "item_photos": [
    {
      "large": "https://images-na.ssl-images-amazon.com/images/I/719022GksTL._SL1500_.jpg",
      "small": "https://images-na.ssl-images-amazon.com/images/I/719022GksTL._SL1500_.jpg",
      "id": 473580,
      "is_main": false
    },
    {
      "large": "http://www.angelfire.com/mech/jinsaotome/Blog/Juggyreview_001.JPG",
      "small": "http://www.angelfire.com/mech/jinsaotome/Blog/Juggyreview_001.JPG",
      "id": 473579,
      "is_main": false
    },
    {
      "large": "https://actionfigurecanada.files.wordpress.com/2011/09/juggernaut-marvel-select-figure-2.jpg",
      "small": "https://actionfigurecanada.files.wordpress.com/2011/09/juggernaut-marvel-select-figure-2.jpg",
      "id": 473581,
      "is_main": true
    },
    {
      "large": "http://4.bp.blogspot.com/-qNdk8wZtZSY/UgGmCHgBLJI/AAAAAAAB0pk/2qpwZTuNSls/s1600/52458-1.jpg",
      "small": "http://4.bp.blogspot.com/-qNdk8wZtZSY/UgGmCHgBLJI/AAAAAAAB0pk/2qpwZTuNSls/s1600/52458-1.jpg",
      "id": 473582,
      "is_main": false
    },
    {
      "large": "https://s-media-cache-ak0.pinimg.com/originals/08/38/8f/08388f9c134afd90f567be95a4fe610b.jpg",
      "small": "https://s-media-cache-ak0.pinimg.com/originals/08/38/8f/08388f9c134afd90f567be95a4fe610b.jpg",
      "id": 473583,
      "is_main": false
    }
  ],
  "recent_comments": [
    {
      "id": 184543,
      "comment": "Miss those classic outfits ",
      "user": {
        "id": 13118,
        "nickname": "colossus",
        "name": "Piotr Rasputin",
        "avatar": {
          "small_url": "http://cdn2-www.comingsoon.net/assets/uploads/2015/12/deadpool-colossus-header.jpg",
          "is_default": false
        },
        "share_url": "http://box.es/nixon"
      }
    },
  ],
}
