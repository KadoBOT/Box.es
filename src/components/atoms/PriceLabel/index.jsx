import React, {PropTypes} from 'react'

import './price-label.css'

const PriceLabel = ({price}) => {
  return(
    <div className="price-label">
      <p className="price-label__price">
        <span className="price-label__price__coin">$</span> {price}
      </p>
    </div>
  )
}

PriceLabel.propTypes = {
  price: PropTypes.number.isRequired
}

export default PriceLabel
