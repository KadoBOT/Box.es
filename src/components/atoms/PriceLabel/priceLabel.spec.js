import React from 'react'
import renderer from 'react-test-renderer';

import PriceLabel from './index'
import templateState from '../../../helpers/_templateState'

it('should render the PriceLabel', () => {
    const wrapper = renderer.create(
      <PriceLabel price={templateState.item.price} />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
});
