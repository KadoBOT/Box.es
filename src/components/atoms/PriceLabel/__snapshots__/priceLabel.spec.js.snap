exports[`test should render the PriceLabel 1`] = `
<div
  className="price-label">
  <p
    className="price-label__price">
    <span
      className="price-label__price__coin">
      $
    </span>
     
    22.43
  </p>
</div>
`;
